### PR TITLE
Content Length issue #355

### PR DESCRIPTION
**Pull request by geoperez**
I changed how the Content-Length is used in Request and Response. Instead of use a private member, they use the same value in Headers collection.

### CODE REVIEW

File: src/EmbedIO/Net/Internal/HttpListenerRequest.cs [hunk 1]
```diff
@@ -304,8 +303,8 @@ internal void FinishInitialization()
             }
 
             CreateQueryString(_url.Query);
-
-            if (!_clSet && (HttpVerb == HttpVerbs.Post || HttpVerb == HttpVerbs.Put))
+            
+            if (ContentLength64 < 0 && (HttpVerb == HttpVerbs.Post || HttpVerb == HttpVerbs.Put))
```
Comment (AbeniMatteo) [inline src/EmbedIO/Net/Internal/HttpListenerRequest.cs]: Shouldn't be `ContentLength64 == 0` or `<=`? @geoperez

Comment (geoperez) [reply src/EmbedIO/Net/Internal/HttpListenerRequest.cs]: Done!

File: test/EmbedIO.Tests/Issues/Issue355_ContentResponseLength.cs [hunk 2]
```diff
@@ -7,16 +7,21 @@ namespace EmbedIO.Tests.Issues
 {
     public class Issue355_ContentResponseLength
     {
-        [Test]
-        public async Task ActionModule_Handle_ContentLengthProperly()
+        [TestCase(true)]
```
Comment (rdeago) [inline test/EmbedIO.Tests/Issues/Issue355_ContentResponseLength.cs]: Two separate tests would be better IMHO. `TestCaseAttribute` should be used to inject data in tests, not to change the behavior of a test.

File: src/EmbedIO/Net/Internal/HttpListenerRequest.cs [hunk 3]
```diff
@@ -325,7 +324,7 @@ internal void AddHeader(string header)
 
             var name = header.Substring(0, colon).Trim();
             var val = header.Substring(colon + 1).Trim();
-            
+
```
Comment (rdeago) [inline src/EmbedIO/Net/Internal/HttpListenerRequest.cs]: Please keep style fixes in separate commits.

File: test/EmbedIO.Tests/Issues/Issue355_ContentResponseLength.cs (hunk 2 above)
Comment (geoperez) [reply test/EmbedIO.Tests/Issues/Issue355_ContentResponseLength.cs]: Done!
